test/main.go

```diff
@@ -205,3 +208,38 @@ func GetDynamicKubernetesClient() (*dynamic.DynamicClient, error) {
 	}
 	return client, nil
 }
+
+func GetUrl(client *http.Client, token string, url string) (*unstructured.UnstructuredList, error) {
+	request, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		fmt.Printf("could not create a request, %s", err)
+		return nil, err
+	}
+	request.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
+	response, err := client.Do(request)
+	if err != nil {
+		fmt.Printf("Couldn't get a response HTTP, %s\n", err)
+		return nil, err
+	}
+	defer response.Body.Close()
+
+	body, err := io.ReadAll(response.Body)
+	if err != nil {
+		fmt.Printf("Couldn't read Body, %s\n", err)
+		return nil, err
+	}
+
+	if response.StatusCode != http.StatusOK {
+		fmt.Printf("The HTTP status returned is not OK, %s - %s \n", response.Status, string(body))
+		return nil, fmt.Errorf("%d", response.StatusCode)
+	}
+
+	var data unstructured.UnstructuredList
+	err = json.Unmarshal(body, &data)
+	if err != nil {
+		fmt.Printf("Couldn't unmarshal JSON, %s\n", err)
+		return nil, err
+	}
+	fmt.Printf("The HTTP status returned is OK, %s, returned %d items\n", response.Status, len(data.Items))
+	return &data, nil
```

Comment:
If this is extracted to a separate function, then `GetUrl` could be used in the `logging_test.go` to make it more compact, readable and easy to maintain

Reply:
I created `getUrl` which has the common functionality, take a look and let me know how you find it.

Reply:
it looks good!